[Sharmaine Cheong] iP

sharmainec opened the pull request:
### Tokio

> 
Tokio remembers things for Rio, so Rio don't have to! It is 
- text-based
- easy to learn
- ~~FAST~~ SUPER FAST to use

Features:
1. Managing tasks
2. Managing deadlines
3. Managing events

---

If you Java programmer, you can use it to practice Java too. Here's the ``main`` method:
```java
public static void main(String[] args) throws IOException {
        Duke duke = new Duke("data/tasks.txt");
        duke.run();
    }
```

Commit:
Add Gradle support

sharmainec committed:
Level 1. Greet, Echo, Exit

sharmainec committed:
Add, List

sharmainec committed:
Level 3. Mark as done.

sharmainec committed:
Level 4. ToDos, Events, Deadlines

sharmainec committed:
A-TextUiTesting: Automated Text UI Testing

sharmainec committed:
Level 5. Handle Errors

sharmainec committed:
Level 6. Delete

sharmainec committed:
print formatting

sharmainec committed:
A-Enums: Use of enums for Instruction types

sharmainec committed:
Level 7. Save.

sharmainec committed:
Level 8. Date

sharmainec committed:
Merge branch 'branch-level-7'

* branch-level-7:
  Level 7. Save.

sharmainec committed:
Merge branch 'branch-level-8'

* branch-level-8:
  Level 8. Date

# Conflicts:
#	src/main/java/Duke.java

sharmainec committed:
A-MoreOOP.

sharmainec committed:
A-Packages: Organize into Packages. New packages include: commands, exceptions, tasks, viper

sharmainec committed:
Update date and time formatting for Deadlines and Events

sharmainec committed:
Add TasksTest.java and CommandsTest.java

James-Kua commented on src/main/java/viper/Instruction.java:
```diff
@@ -0,0 +1,29 @@
+    EMPTY("empty"),
+    ;
+
+    private final String s;
```
Should this variable have a more descriptive name to let the user know what this variable is storing?

James-Kua commented on src/main/java/viper/Storage.java:
```diff
@@ -0,0 +1,149 @@
+        }
+        
+        fw.close();
+        boolean success = tempFile.renameTo(file);
```
Should this boolean have a more intuitive name like isRenameSuccess?

James-Kua commented on src/main/java/viper/Duke.java:
```diff
@@ -0,0 +1,47 @@
+
+    public void run() {
+        ui.showWelcome();
+        boolean isExit = false;
+        while (!isExit) {
```
Would renaming boolean isRunning = true, followed by while (isRunning) be more intuitive?

James-Kua reviewed (commented):
Overall, I thought that the code was very neat and tidy and the naming of variables were also very intuitive which made it easy for me to understand what was happening as I was going through your code. Each function also has a clear sole purpose. Nice!

sharmainec committed:
Update JavaDocs for public classes and methods.

sharmainec committed:
Update codes to comply with coding standard

sharmainec committed:
Add FindCommand.java to support finding of keyword in tasks

sharmainec committed:
Update tasks.txt

sharmainec committed:
Merge branch 'branch-A-JavaDoc'

* branch-A-JavaDoc:
  Update JavaDocs for public classes and methods.

# Conflicts:
#	data/tasks.txt

sharmainec committed:
Merge branch 'branch-A-CodingStandard'

* branch-A-CodingStandard:
  Update codes to comply with coding standard

sharmainec committed:
Merge branch 'branch-Level-9'

* branch-Level-9:
  Add FindCommand.java to support finding of keyword in tasks

# Conflicts:
#	src/main/java/viper/Ui.java

sharmainec committed:
Merge commit 'd8398594b7bc43da5eb865321c5a50cec4b3923d' into add-gradle-support

* commit 'd8398594b7bc43da5eb865321c5a50cec4b3923d':
  Add Gradle support

sharmainec committed:
Update files to follow CheckStyle format.

sharmainec committed:
Merge branch 'branch-A-Checkstyle'

* branch-A-Checkstyle:
  Update files to follow CheckStyle format.

sharmainec committed:
Update file structure.

sharmainec committed:
Added basic GUI for duke.

sharmainec committed:
Update GUI and check style.

JosephN37 commented on src/main/java/tokio/Duke.java:
```diff
@@ -0,0 +1,104 @@
+        duke.run();
+    }
+
+    /**
+     * You should have your own function to generate a response to user input.
+     * Replace this stub with your completed method.
+     */
+    public String getResponse(String input) {
+        try {
+            Command command = Parser.parse(input);
+            if (command.isExit()) {
+                ui.printBye();
+                PauseTransition delay = new PauseTransition(Duration.seconds(2));
+                delay.setOnFinished(event -> Platform.exit());
+                delay.play();
+            }
+            return command.execute(tasks, ui, storage);
+        } catch (DukeException | IOException e) {
+            return e.getMessage();
+        }
+    }
```
A friendly reminder to update the javadoc for this method :)

JosephN37 commented on src/main/java/tokio/commands/AddCommand.java:
```diff
@@ -0,0 +1,70 @@
+        this.cmdLine = cmdLine;
+    }
+
+    @Override
+    public String execute(TaskList tasks, Ui ui, Storage storage) {
```
Maybe you can make this function easier to be understood by providing javadocs, though the name might be self-explanatory to some extent

sharmainec committed:
Update to generate new JAR file

sharmainec committed:
Update UI.java to return messages with the proper formatting

sharmainec committed:
Merge branch 'branch-level-10-GUI'

* branch-level-10-GUI:
  Update UI.java to return messages with the proper formatting

# Conflicts:
#	src/main/java/tokio/ui/Ui.java

sharmainec committed:
Update codes with assert

sharmainec committed:
Improve code quality

sharmainec committed:
Merge pull request #1 from sharmainec/branch-A-Assertions

Include assertions

sharmainec committed:
Merge branch 'master' of https://github.com/sharmainec/ip

* 'master' of https://github.com/sharmainec/ip:
  Update codes with assert

sharmainec committed:
Merge branch 'branch-A-CodeQuality'

* branch-A-CodeQuality:
  Improve code quality

# Conflicts:
#	src/main/java/tokio/commands/DeleteCommand.java
#	src/main/java/tokio/commands/DoneCommand.java

sharmainec committed:
Update with check for duplicate tasks

sharmainec committed:
Update Javadocs

sharmainec committed:
Merge branch 'branch-C-DetectDuplicates'

* branch-C-DetectDuplicates:
  Update Javadocs
  Update with check for duplicate tasks

sharmainec committed:
Add formatToStorage() method for Task and its subclasses

Tasks have been previously formatted using the overridden toString(), however, it is not user-friendly
because users might find it difficult to read between "X" and " ".

Separating the toString() method into 2 methods, one for user display and another for storage
will solve the issue.

Let's,
* Update toString() method to be used only for user display
* Update task status icon to a tick and a cross
* Add new method formatToStorage() to format tasks for storage purposes only

sharmainec committed:
Improve the GUI for Tokio

sharmainec committed:
Update tests for Commands and Tasks

Previously AddCommand handles the adding of Todo, Deadline and Event. Now, there's 3 separate
classes handling each type of task, so the new test cases would test for these new classes.

Previously, the test for tasks only included tasks which were undone. Now, tests cases includes
both done and undone tasks.

sharmainec committed:
Add Ui.png

sharmainec committed:
Add icon for main window and update error message

sharmainec committed:
Update Ui.png

sharmainec committed:
Set theme jekyll-theme-cayman

sharmainec committed:
Update codes for ui and find method

sharmainec committed:
Add images for user guide

sharmainec committed:
Update image for user guide

sharmainec committed:
Update README.md

sharmainec committed:
Update README.md 

Replaced HTML snippets

sharmainec committed:
Update README.md

Minor changes to formatting

sharmainec committed:
Update README.md

sharmainec committed:
Update header comments

sharmainec committed:
Update check for duplicate tasks to be case insensitive

sharmainec committed:
Revert "Update check for duplicate tasks to be case insensitive"

This reverts commit cc11e346125194e4a938299afabf076e4e920c7b.

sharmainec committed:
Update check for duplicate tasks to be case insensitive

sharmainec committed:
Update parser

sharmainec committed:
Update README.md

Find is case insensitive
